patches/api/0446-ChunkStatusChangeEvent.patch

```diff
@@ -0,0 +1,71 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Wed, 8 Nov 2023 09:23:23 -0600
+Subject: [PATCH] ChunkStatusChangeEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/chunk/ChunkStatusChangeEvent.java b/src/main/java/com/destroystokyo/paper/event/chunk/ChunkStatusChangeEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..9a556fe4321ad1b4809c0870a484124284239a7f
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/chunk/ChunkStatusChangeEvent.java
+@@ -0,0 +1,59 @@
++package com.destroystokyo.paper.event.chunk;
++
++import org.bukkit.Chunk;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when the Full Status of a Chunk changes. This cannot be modified; you wouldn't want to if you tried.
++ * Chunk Statuses are a dark magic and this event only exposes the changes of status and will probably do nothing else.
++ */
++public class ChunkStatusChangeEvent extends Event {
++
++    private static final HandlerList handlers = new HandlerList();
++    private final Chunk chunk;
++    private final ChunkStatus currentState;
++    private final ChunkStatus newState;
++
++    public ChunkStatusChangeEvent(Chunk chunk, ChunkStatus currentState, ChunkStatus newState) {
++        super();
++        this.chunk = chunk;
++        this.currentState = currentState;
++        this.newState = newState;
++    }
++
++    public Chunk getChunk() {
++        return chunk;
++    }
++
++    public ChunkStatus getCurrentState() {
++        return currentState;
++    }
++
++    public ChunkStatus getNewState() {
++        return newState;
++    }
++
++    public boolean isUpgrade() {
++        return newState.ordinal() > currentState.ordinal();
++    }
++
++    @Override
++    @NotNull
++    public HandlerList getHandlers() {
++        return handlers;
++    }
++
++    @NotNull
++    public static HandlerList getHandlerList() {
++        return handlers;
++    }
++
++    public enum ChunkStatus {
++        INACCESSIBLE,
++        FULL,
++        BLOCK_TICKING,
++        ENTITY_TICKING;
++    }
```

Comment:
Cant we use Chunk$LoadLevel?

Reply:
kinda, but they messed up the order of them so it breaks the nice ordinal thing

Reply:
Yeah, was nice for the isUpgrade() method